src/components/ConnectWalletModal/ConnectWalletModal.jsx

```diff
@@ -0,0 +1,41 @@
+import { useAccount, useConnectors } from "@starknet-react/core";
+import Modal from "components/Modal/Modal";
+import { useEffect } from "react";
+
+// type Props = {
+//   walletModalVisible: boolean;
+//   setWalletModalVisible: (visible: boolean) => void;
+// };
+
+export function ConnectWalletModal({ walletModalVisible, setWalletModalVisible } /*: Props */) {
+  const { address } = useAccount();
+  const { connect, connectors } = useConnectors();
+
+  //   const userOnMobileDevice = "navigator" in window && isMobileDevice(window.navigator);
+
+  useEffect(() => {
+    if (address !== undefined) {
+      setWalletModalVisible(false);
+    }
+  }, [address, setWalletModalVisible]);
```

Comment:
Change this into a derived state
```suggestion
 const isAddressDefined = address !== undefined;
```

And then you use that as part of the state to show the modal

Reply:
For later